sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1072,8 +1072,17 @@ class SessionCatalog(
   def functionExists(name: FunctionIdentifier): Boolean = {
     val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
     requireDbExists(db)
-    functionRegistry.functionExists(name) ||
-      externalCatalog.functionExists(db, name.funcName)
+    builtinFunctionExists(name) || externalFunctionExists(name)
+  }
+
+  def builtinFunctionExists(name: FunctionIdentifier): Boolean = {
```

Comment:
All the functions in functionRegistry are built-in?

Reply:
@gatorsmile sorry for the delay. Not all the functions in functionRegistry are built-in, there could be UDF. I am thinking to change `builtinFunctionExists` to `registryFunctionExist`, what do you think? I also need to change the PR description if we just cache the external function names. Thanks.

Reply:
Let us do not add these APIs to SessionCatalog. Just create private functions?

Reply:
ok, sure.

Reply:
```Scala
  /**
   * Returns whether this function has been registered in the function registry of the current
   * session. If not existed, returns false.
   */
  def isRegisteredFunction(name: FunctionIdentifier): Boolean = {
    functionRegistry.functionExists(name)
  }

  /**
   * Returns whether it is a persistent function. If not existed, returns false.
   */
  def isPersistentFunction(name: FunctionIdentifier): Boolean = {
    val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
    databaseExists(db) && externalCatalog.functionExists(db, name.funcName)
  }
```

Reply:
Please also add the unit test cases to `SessionCatalogSuite`

Reply:
Move them after `def isTemporaryFunction(name: FunctionIdentifier): Boolean `

Reply:
@gatorsmile Sorry for the delay. I was working on something else. Thanks very much for the help, I will add these two new APIs in `SessionCatalog` and add test cases in `SessionCatalogSuite`